src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1614,9 +1614,10 @@ public static XPathReference getAbsRef(XPathReference ref, TreeReference parentR
             tref = TreeReference.selfRef(); //only happens for <group>s with no binding
         }
 
+        TreeReference refPreContextualization = tref;

```

Comment:
tref gets redefined below which is what ends up making its value null if the contextualization fails. This just retains it's "pre-contextualized" state, which is generally a relative reference  like ("../something") which didn't work